formats/Filtered/libs/ext.srf.filtered.value-filter.js

```diff
@@ -254,11 +262,14 @@
 				filtercontrols.height( height );
 			}
 
-
 			// insert options (checkboxes and labels) and attach event handlers
 			// TODO: Do we need to wrap these in a form?
-			for ( var j in sortedDistinctValues ) {
+			for (var j = 0; j < sortedDistinctValues.length; j++) {
 				var option = $('<div class="filtered-value-option">');
+
+				// Try to get label, if not fall back to value id
+				var label = distinctValues[sortedDistinctValues[j]] || sortedDistinctValues[j];
+
 				var checkbox = $('<input type="checkbox" class="filtered-value-value" value="' + sortedDistinctValues[j] + '"  >');
```

Comment:
Mea culpa re the coding style. I know it's a mess. It also has a number of design issues and if I had the time I would rewrite the whole thing from scratch.
@Fannon I'll have a look at this asap.